File: web-app/src/repositories/dareRepository.ts
import { collection, query, where } from "firebase/firestore";
import { db } from "../firebase";
import { fetchCollection } from "../lib/firestore";
import type { Dare } from "../types";

export async function fetchActiveDares(): Promise<Dare[]> {
  const q = query(collection(db, "dares"), where("active", "==", true));
  return fetchCollection<Dare>(q);
}

File: ios-app/OutAndAbout/Core/Repositories/DareRepository.swift
import Foundation
import FirebaseFirestore

protocol DareRepositoryProtocol {
    func fetchDares() async throws -> [Dare]
}

final class DareRepository: DareRepositoryProtocol {
    private let firestore = FirestoreService.shared

    func fetchDares() async throws -> [Dare] {
        let query = firestore.db.collection("dares")
            .whereField("active", isEqualTo: true)
            .order(by: "category")
        let snapshot = try await query.getDocuments()
        return try snapshot.documents.map { try $0.data(as: Dare.self) }
    }
}
